### DISCUSSION TITLE
Nginx Proxy Manager - Reverse proxy throughput is very low

### DISCUSSION THREAD

**Discussion by kristof-dragon** [tteck/Proxmox#2806] [Help Needed]
Hi,

The Proxmox LXC container created by your script results in a very slow throughput reverse proxy (for me).
I was trying to see if anyone else had similar issue, and if yes, have they solved it or what I should be looking for to mitigate the throughput issue.
As I have tested it directly and via the created NPM LXC, the speed via the container is very slow compared to a direct test.
(Openspeed docker and tested it through the reverse proxy. When testing directly in the local network I get ~2.2Gbps up and down with some fluctuation, but when testing through the reverse proxy, I get around 5-8Mbps (up and down).)

Any help would be appreciated.
Thank you in advance

======Original main post - rephrased it.. as per above===========
<s>I have set up the debian based NPM docker with the [script on Proxmox](https://tteck.github.io/Proxmox/), configured the proxy hosts, and noticed that Nextcloud was slow when accessing it via the nginx reverse proxy.
I have also set up an Openspeed docker and tested it through the reverse proxy. When testing directly in the local network I get ~2.2Gbps up and down with some fluctuation, but when testing through the reverse proxy, I get around 5-8Mbps (up and down).

The same bottlenecking is true for all services (including Nextcloud file upload and sync)

Could someone help me, what I would need to tweak to get rid of the speed restriction (or at least increase the speed)
</s>
Thank you in advance

**Comment by tteck** [comment]
> NPM-Alpine docker with the script on Proxmox

What script are you talking about?

**Reply by kristof-dragon** [reply]
I meant your script from https://tteck.github.io/Proxmox/

this script:
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/nginxproxymanager.sh)"

**edited original post to correct the "alpine" to "Debian" as the script creating the NPM docker uses debian. My mistake.

**Reply by tteck** [reply]
That script has nothing to do with Alpine or Docker. (I see that you edited your post.)

**Reply by kristof-dragon** [reply]
I should probably try this again.... (explaining the situation better)
The Proxmox LXC container created by your script results in a very slow throughput reverse proxy (for me).
I was trying to see if anyone else had similar issue, and if yes, have they solved it or what I should be looking for to mitigate the throughput issue.
As I have tested it directly and via the created NPM LXC, the speed via the container is very slow compared to a direct test.

Any help would be appreciated.

**Reply by kristof-dragon** [reply]
@tteck: have you experienced anything similar?

**Reply by kristof-dragon** [reply]
Update:
If I turn off all toggles on the "SSL" tab on the proxy host setting (Force SSL, HTTP/2 Support..), the speed climbs up to ~30Mb/s down and ~20Mb/s up... which is better, but still nowhere near the acceptable level. This is on a 4c i5-8365U cpu with AES-NI enabled (in comparison testing the npm docker on a single board computer (OrangePiZero3 4core Cortex-A53@1.5Ghz gives the exact same speeds...)

**Reply by Musrha** [reply]
Can you try to create another NPM LXC and try there? I installed openspeedtest in my alpine-docker and checked from my local PC by direct IP and then by NPM proxy (test.mydomain.com) and it was the same speeds in both cases (980~ both upload and download)
